### DISCUSSION TITLE
Questions on Tet10 Element！

### DISCUSSION THREAD

**Discussion by QuatorzeM** [OpenRadioss/OpenRadioss#1514] [Q&A] [answered]
Dear Community,

Salut !

Recently, I am working on the code of TET10 in Openradioss. I come to understand a little the process in this part but I still have some theoretical questions to discuss:
1. I see there is a parameter for the special TET10 element type with 5 intergration point. I just wondering what this element type is used for ? Is it designed for the element with low quality (like thin element which has a very small characteristic length)? 
2. For the incompressible problem in plastic,  I see it is ANP method used, but it seems not the same in TET4? 
3. I see there is 4 subroutine to calculate the determinant in S10Jacob.F (S10JACOB, S10JACOB1, S10JACOBT, S10JACOBNEL), what is the difference？ 

Thanks :)  
Fabien

**Comment by qiangRadioss** [comment]
Hello Fabien,

thanks for your interest on TET10 element, see the following answers:
1. actual implementation has fixed the number of integration point to 4, we don't know neither which kind of applications need 5 integration points, please share with us if you knew something.
2. not being sure what you called "ANP" method, do you mean "Smoothing nodal pressure"? we could do it both with TET4 and TET10, just it's optional with TET4 (Itetra4=3), and automatically being applied on TET10 in function laws used
3. yes, the 4 subroutines you mentioned are the same of calculating the Jacobian, the original is S10JACOB, the differences for others come from the different configurations used (nodal co-ordinates) with special options: 
   - S10JACOB1 is used for option /NEGVOL (see Engine input manual), some elements use fixed configuration once switched to small strain
   - S10JACOBT is used for total strain cases Ismstr=10,12, some data are based on the initial configuration, then aren't to be computed in each cycle (e.g. JAC_I) to save computation times
   - S10JACOBNEL is used for case Ismstr=12 where some elements are switched to small strain   

hope these help

Qiang.

**Reply by QuatorzeM** [reply]
Hello Qiang, 

Thanks a lot for your so kind response. That's very helpful.

Ok, so I will not bother myself with the 5 intergration point TET10 and focus on the normal one. If I find some thing new, I will tell you :）
ANP means 'Average Nodal Pressure', I think it’s the same as "Smoothing Nodal Pressure". I see it's applied for plastic material. 
Thanks for the explanation of the Jacobian subroutines, I will focus more on S10JACOB.  

I still have some questions:
1. is there any special treatment for negative volume TET10 or the TET10 with low element quality, for example the very thin element, which leads to a very small time step?
2.  I see that in subroutine forint.F,  S10FORC3 is called first then the SECTIOS4 is called. The S10FORC3 is used to calculate the internal force, which I can understand, but what is SECTIOS4 used for? Is it a kind of modification like hourglass? It seems the internal force direction is changed in SECTIOS4. 

Best wishes ！
Fabien MENG

**Reply by qiangRadioss** [reply, marked as answer]
Hi Fabien,

For 1. generally speaking you can use the options which allow to switching small strain (e.q. /DT/BRICK/CST/1) or to deleting element (e.g. /DT/BRICK/DEL/1) and combined with mass scaling (e.q. DT/NODA/CST) for time step issue, Itetra10=2 is also a good option to increase time step for tetra10.
For 2. SECTIOS4 is the routine to compute section resulting forces&moments (see /SECT in manual) for output 

Qiang.

**Reply by QuatorzeM** [reply]
Hi Qiang, Thanks a lot! That's helpful. I wiil try to keep this discussion active in the following days :）
Have a nice day!

**Reply by QuatorzeM** [reply]
Hello Qiang,

I have a question in the subroutine S10DEFO3. I see first the velocity strain is calculated DXX DYY DZZ DXY DXZ DYX... then for the case JHBE>=2, it seems the Almansi strain tensor is calculated. 

I want to know what the flag JHBE means? and the function of  Almansi strain tensor， is it used to increase the stability？ 

Thanks，

Fabien

**Comment by qiangRadioss** [comment]
Hi Fabien,

just forget JHBE=2, obsolete hide option and will be removed.
thanks.
Qiang.

**Reply by QuatorzeM** [reply]
Thanks Qiang，that saves a lot of  time. Have a nice weekend!
